internal/engine/eval/rego/lib.go

```diff
@@ -328,6 +428,99 @@ func fileLsHandleDir(path string, bfs billy.Filesystem) (*ast.Term, error) {
 		ast.NewArray(files...)), nil
 }
 
+// FileArchive packages a set of files form the the specified directory into
+// a tarball.  It takes one argument: a list of file or directory paths to
+// include, and outputs the tarball as a string.
+// It's exposed as 'file.archive`.
+func FileArchive(res *interfaces.Result) func(*rego.Rego) {
+	return rego.Function1(
+		&rego.Function{
+			Name: "file.archive",
+			Decl: types.NewFunction(types.Args(types.NewArray(nil, types.S)), types.S),
+		},
+		fsArchive(res.Fs),
+	)
+}
+
+// BaseFileArchive packages a set of files form the the specified directory
+// in the base filesystem (from a pull_request or other diff context) into
+// a tarball.  It takes one argument: a list of file or directory paths to
+// include, and outputs the tarball as a string.
+// It's exposed as 'base_file.archive`.
+func BaseFileArchive(res *interfaces.Result) func(*rego.Rego) {
```

Comment:
Do we want to expose this function to non PR/diff contexts too? I think it would be handy in general.

Reply:
See `FileArchive`, above.  😁